pkg/disttask/framework/scheduler/nodes.go

```diff
@@ -185,3 +185,30 @@ func filterByScope(nodes []proto.ManagedNode, targetScope string) []string {
 	}
 	return nodeIDs
 }
+
+// SplitItemsEvenlyToWorkers split the items into batches that is suitble to be handled by `workerCnt` workers.
+func SplitItemsEvenlyToWorkers[T any](
+	items []T,
+	workerCnt int,
+	defaultStep int,
+	handleOneBatch func([]T),
+) {
```

Comment:
I prefer to make this func only handles `split the items into some batches`, i.e., this function should return `[][]T` out, and let outer logic call `handleOneBatch`